src/connector/src/source/nexmark/source/generator.rs

```diff
@@ -43,7 +44,12 @@ impl NexmarkEventGenerator {
     #[try_stream(ok = Vec<SourceMessage>, error = anyhow::Error)]
     pub async fn into_stream(mut self) {
         loop {
-            yield self.next().await?
+            let chunk = self.next().await?;
+            if chunk.is_empty() {
+                yield pending().await;
```

Comment:
This could be CPU intensive if there are no remaining records to generate, which is problematic with madsim. 🤣